src/dataframe/dataframe.jl

```diff
@@ -1118,32 +1141,33 @@ function Base.push!(df::DataFrame, row::Union{AbstractDict, NamedTuple}; columns
     if !(columns in (:equal, :intersect))
         throw(ArgumentError("`columns` keyword argument must be `:equal` or `:intersect`"))
     end
-    if ncol(df) == 0 && row isa NamedTuple
+    nrows, ncols = size(df)
+    targetrows = nrows + 1
+    if ncols == 0 && row isa NamedTuple
         for (n, v) in pairs(row)
             setproperty!(df, n, fill!(Tables.allocatecolumn(typeof(v), 1), v))
         end
         return df
     end
-    i = 1
     # Only check for equal lengths, as an error will be thrown below if some names don't match
-    if columns === :equal && length(row) != size(df, 2)
+    if columns === :equal && length(row) != ncols
         # TODO: add tests for this case after the deprecation period
         Base.depwarn("In the future push! will require that `row` has the same number" *
                       "of elements as is the number of columns in `df`." *
                       "Use `columns=:intersect` to disable this check.", :push!)
     end
-    for nm in _names(df)
-        try
-            push!(df[!, i], row[nm])
-        catch
-            #clean up partial row
-            for j in 1:(i - 1)
-                pop!(df[!, j])
-            end
-            msg = "Error adding value to column :$nm."
```

Comment:
This was probably useful to understand what is the problematic column. Not sure how we can convey both this information, and the full stacktrace for the actual error.

Reply:
done

Reply:
Clever.